docs/frontend/testing/using-factories-to-generate-data-for-frontend-tests.md

```diff
@@ -0,0 +1,235 @@
+
+```javascript
+  const MY_OBJECT_FIELDS = {
+    ID: 'id',
+    MY_FIELD: 'myField',
+  }
+```
+
+Create a const called `{object}Factory`, giving it a camelCase name. This const is a function that
+
+- takes params
+- returns a `baseFactory`, into which is passed an object containing
+  - `fields`: the fields your factory returns
+  - any other configuration passable to the `build()` functions [here](https://www.npmjs.com/package/@jackfranklin/test-data-bot), such as `postBuild` or `traits`.
+  - the spread params: `...params`.
+
+Export your factory by default.
+
+```javascript
+  const myObjectFactory = (params) => {
+    return baseFactory({
+      [BASE_FIELDS.FIELD]: {
+        [MY_OBJECT_FIELDS.ID]: '...',
+        [MY_OBJECT_FIELDS.MY_FIELD]: '...',
+      },
+      // optional postBuild and traits...,
+      ...params
+    })
+  }
+```
+
+### Build and post-build
+
+Factories generate data in two phases, the build and the post-build. Most values can be generated in the build phase. Use postBuild when the value of one field should be based on the value of another generated value.
+
+### Fields
+
+The `fields` object defines the structure and values of the object your factory will return. This happens at build time. Values of this object can be literals, functions, or another object containing either.
+
+```javascript
+  [BASE_OBJECT.FIELDS]: {
+    [OBJECT_FIELDS.FIELD_1]: 'value',
+    [OBJECT_FIELDS.FIELD_2]: myCoolFunction,
+    [OBJECT_FIELDS.FIELD_3]: {
+      [SUBOBJECT_FIELDS.FIELD_1]: 'value',
+    }
+  }
+```
+
+#### Literals
+
+If your field's value is a literal, the factory will always return that value. For example:
+
+`[OBJECT_FIELDS.FIELD_1]: 'value'`
+
+will always set the value of `[OBJECT_FIELDS.FIELD_1]` to `'my value'` by default.
+
+#### Functions
+
+Functions will be evaluated and the field set to their value. This is handy when a hardcoded value shouldn't be set, which is usually the case. You might generate an ID with a function, for example.
+
+Note: pass in the name of a function or an anoymous function. The builder will handle executing the function for you.
+
+`[OBJECT_FIELDS.FIELD_2]: myCoolFunction,`
+
+or
+
+`[OBJECT_FIELDS.FIELD_2]: () => 'my cool return value',`
+
+#### Objects
+
+Fields can be set to objects and their values will be set recursively.
+
+```javascript
+  [OBJECT_FIELDS.FIELD_3]: {
+    [SUBOBJECT_FIELDS.FIELD_1]: 'value',
+  }
+```
+
+### Subfactories
+
+If your object has fields that themselves can be generated by a factory, you can pass in a factory function as the value of that field:
+
+`[OBJECT_FIELDS.SUBOBJECT_FIELD]: (args) => subobjectFactory(args)`
+
+Be sure not to call the function, or its override features will not work.
```

Comment:
**suggestion (non-blocking)**: It might be good to call out that for subfactories, only overrides work. We don't currently pass traits along as far as I can tell.

Reply:
I added a note there. That might be something worth implementing in the future, though I imagine its use case would be pretty narrow.

Reply:
Yeah, we can wait to see if it's a thing that seems useful.